src/v2/Apps/Fair/Components/FairExhibitors/FairExhibitorCard.tsx

```diff
@@ -0,0 +1,174 @@
+    }
+
+    const visibleCities = cities.slice(0, VISIBLE_CITIES_NUM).join(", ")
+
+    if (cities.length > VISIBLE_CITIES_NUM) {
+      return `${visibleCities}, +${cities.length - VISIBLE_CITIES_NUM} more`
+    }
+
+    return visibleCities
+  }
+
+  return (
+    <Box>
+      <RouterLink
+        to={partner.href}
+        noUnderline
+        onClick={() => tracking.trackEvent(tappedPartnerTrackingData)}
+      >
+        <Flex mb={1} flex={1}>
+          <BorderBox width={52} height={52} p={0} mr={1}>
+            {profile?.icon?.cropped && (
+              <img
+                src={profile?.icon?.cropped?.src}
+                srcSet={profile?.icon?.cropped?.srcSet}
+                alt={`Logo of ${name}`}
+                width={50}
+                height={50}
+              />
+            )}
+          </BorderBox>
+          <Box overflow="hidden">
+            <Text variant="md" overflowEllipsis>
+              {name}
+            </Text>
+            {locations?.totalCount && locations.edges ? (
+              <Text variant="md" color="black60" overflowEllipsis>
+                {partnerAddress()}
+              </Text>
+            ) : null}
+          </Box>
+          {partner.profile && (
+            <Box order={2} ml="auto">
+              <FollowProfileButton
+                profile={partner.profile}
+                user={user}
+                contextModule={ContextModule.partnerHeader}
+                buttonProps={{
+                  size: "small",
+                  variant: "secondaryOutline",
+                  width: 70,
+                  height: 30,
+                }}
+              />
+            </Box>
+          )}
+        </Flex>
+      </RouterLink>
+
+      <Media greaterThan="xs">
+        <RouterLink
+          to={partner.href}
+          noUnderline
+          onClick={() => tracking.trackEvent(tappedPartnerTrackingData)}
+        >
+          <BorderBox p={0}>
+            <ResponsiveBox aspectWidth={400} aspectHeight={250} maxHeight={400}>
+              {profile?.image?.url ? (
+                <Image
+                  width="100%"
+                  height="100%"
+                  src={profile.image.url}
+                  alt="Please make sure you describe the image"
+                />
+              ) : null}
+            </ResponsiveBox>
+          </BorderBox>
+        </RouterLink>
+      </Media>
+    </Box>
+  )
+}
+
+export const FairExhibitorCardFragmentContainer = createFragmentContainer(
+  FairExhibitorCard,
+  {
+    partner: graphql`
+      fragment FairExhibitorCard_partner on Partner {
+        name
+        href
+        internalID
+        slug
+        profile {
+          ...FollowProfileButton_profile
+          icon {
+            cropped(width: 50, height: 50) {
+              src
+              srcSet
+            }
+          }
+          image {
+            url(version: "medium")
+          }
+        }
+        locations: locationsConnection(first: 20) {
```

Comment:
**suggestion**: I believe the partner type already has a `cities` field which is a unique list of cities the partner has locations within.